src/lib.rs

```diff
@@ -800,6 +800,15 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 		self.event_queue.next_event()
 	}
 
+	/// Returns the next event in the event queue.
+	///
+	/// Will asynchronously poll the event queue until the next event is ready.
+	///
+	/// **Note:** this will always return the same event until handling is confirmed via [`Node::event_handled`].
+	pub async fn next_event_async(&self) -> Event {
```

Comment:
its worth following the naming convention with `wait_next_event` and call this `async_next_event`
also, Its a bit confusing to have `wait` and `async` as usually they both mean async something..
maybe wait_next_event should be sync_next_event?

Reply:
Thanks for the explanation. 

I would also go with `poll_` , would look better than `async fn name_async()`. 
`future_next_event` could be another option..